src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -102,10 +97,10 @@ public override void Contextualize(ViewContext viewContext)
                 throw new ArgumentNullException(nameof(expression));
             }
 
-            var modelExplorer = GetModelExplorer(expression);
+            var modelExpression = GetModelExpression(expression);
```

Comment:
FYI we designed `ModelExpressionProvider` for tag helpers, long after `HtmlHelper<T>` was in the new world. This new usage looks good.